pilot/pkg/bootstrap/server.go

```diff
@@ -319,6 +319,7 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 		if err != nil {
 			return nil, fmt.Errorf("error initializing sidecar injector: %v", err)
 		}
+		s.readinessFlags.sidecarInjectorReady.Store(true)
```

Comment:
YOu should add 
if wh == nil{
    s.readinessFlags.sidecarInjectorReady.Store(true)
}

Reply:
Fairly certain wrapping the readiness in the suggested conditional will prevent the cases where the injector is used from ever becoming ready as this is the only location where the readiness is marked as true.